object_db.go

```diff
@@ -159,6 +159,7 @@ func (o *ObjectDatabase) WriteBlob(b *Blob) ([]byte, error) {
 		return nil, err
 	}
 	defer os.Remove(buf.Name())
+	defer buf.Close()
```

Comment:
This certainly seems correct to me, although I think that grokking it requires that the reader know that defer statements are executed in LIFO order (as you pointed out in your commit message). One way to potentially make this clearer is:

```go
defer func() {
        os.Remove(buf.Name())
        buf.Close()
}()
```

Which makes it clearer to see that the handle is removed and _then_ the buffer is closed. But, I think that we would take care of both of the call-sites below by writing something like:

```go
func (o *ObjectDatabase) cleanup(f *os.File) {
        os.Remove(f.Name())
        f.Close()
}
```

and then writing:

```go
defer o.cleanup(buf) // <- on L162
defer o.cleanup(tmp) // <- on L241
```

##

Of course, we could go even further and wrap the result of `io/ioutil.TempFile` in our own type which handles the cleanup for us by implementing `io.ReadCloser`, but I think that may be going overkill for what we need to do here.

Reply:
Sure, I think that would be a good improvement.